sdks/java/io/cdc/src/main/java/org/apache/beam/io/cdc/DebeziumSDFDatabaseHistory.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.io.cdc;
+
+import io.debezium.document.Document;
+import io.debezium.document.DocumentReader;
+import io.debezium.document.DocumentWriter;
+import io.debezium.relational.history.AbstractDatabaseHistory;
+import io.debezium.relational.history.DatabaseHistoryException;
+import io.debezium.relational.history.HistoryRecord;
+import org.apache.beam.sdk.transforms.splittabledofn.RestrictionTracker;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class DebeziumSDFDatabaseHistory extends AbstractDatabaseHistory {
+    private static final Logger LOG = LoggerFactory.getLogger(DebeziumSDFDatabaseHistory.class);
+
+    private List<byte[]> history;
+
+    public DebeziumSDFDatabaseHistory() {
+        this.history = new ArrayList<>();
+    }
+
+    @Override
+    public void start() {
+        super.start();
+        LOG.info("------------ STARTING THE DATABASE HISTORY! - trackers: {} - config: {}",
+                KafkaSourceConsumerFn.restrictionTrackers, config.asMap());
+
+        // TODO(pabloem): Figure out how to link these. For now, we'll just link directly.
+        RestrictionTracker<DebeziumOffsetHolder, ?> tracker = KafkaSourceConsumerFn.restrictionTrackers.get(
+                // JUST GETTING THE FIRST KEY. This will not work in the future.
+                KafkaSourceConsumerFn.restrictionTrackers.keySet().iterator().next());
```

Comment:
have you had a chance to run this on Dataflow? I expect that this may not work well on Dataflow.